### PR TITLE
Amendment cleanups

### PR DESCRIPTION
**Pull request by seelabs**
The first commit has already been reviewed. Only the two "cleanup" commits need to be reviewed

### CODE REVIEW

**Comment by seelabs** [comment]
Force pushed: added a test case and some other minor cleanups (the two original commits are unchanged, so if they have already been reviewed they don't need to be reviewed again).

File: src/test/app/Flow_test.cpp [hunk 1]
```diff
@@ -1157,11 +1157,57 @@ struct Flow_test : public beast::unit_test::suite
...
+
+        env(offer(carol, XRP(1), bob["USD"](1000)));
+        env.close();
+
+        env(pay(alice, alice, USD(1000)), path(~bob["USD"], bob, gw),
+            sendmax(XRP(1)), txflags(tfNoRippleDirect),
+            ter(withFix ? tecPATH_DRY : tesSUCCESS));
+        env.close();
+
+        pass();
```
Comment (ximinez) [inline src/test/app/Flow_test.cpp]: No need to call `pass()` here, since there are checks done by the various `env` functions.

**Comment by seelabs** [comment]
Squashed

**Comment by nbougalis** [comment]
Merged as 3bd9772c04bedcf618ea3f3dfb64ee101e4b45b5 and 068889e5b17b184a2a22a9e2092aa9260ae6e271

**Pull request closed without merging**